aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -639,18 +639,15 @@ void mse_kernel(TensorIterator& iter) {
 }
 
 void fmod_kernel(TensorIterator& iter) {
-  // Use the dtype of the first argument to retain BC,
-  // change to common_dtype for type promotion in the future
-  // Issue #47779: https://github.com/pytorch/pytorch/issues/47779
-  if (isIntegralType(iter.dtype(), /*includeBool=*/ false)) {
-    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "fmod_cpu", [&]() {
+  if (isIntegralType(iter.common_dtype(), /*includeBool=*/ false)) {
```

Comment:
What happens if fmod is called with a boolean argument?

Reply:
Thanks for mentioning it! I was wondering if anyone really needs a boolean tensor `fmod` to another boolean value. But I will switch it to support this behavior since `numpy` is supporting this.